kie-remote/kie-remote-services/src/main/java/org/kie/remote/services/rest/jaxb/DynamicJaxbContextManager.java

```diff
@@ -0,0 +1,209 @@
+package org.kie.remote.services.rest.jaxb;
+
+
+import static org.kie.remote.services.rest.jaxb.DynamicJaxbContext.contextsCache;
+import static org.kie.remote.services.rest.jaxb.DynamicJaxbContextFilter.DEFAULT_JAXB_CONTEXT_ID;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+
+import org.jbpm.kie.services.impl.KModuleDeploymentUnit;
+import org.jbpm.services.api.DeploymentEvent;
+import org.jbpm.services.api.model.DeploymentUnit;
+import org.jbpm.services.cdi.Deploy;
+import org.jbpm.services.cdi.Undeploy;
+import org.kie.remote.services.exception.KieRemoteServicesDeploymentException;
+import org.kie.remote.services.exception.KieRemoteServicesInternalError;
+import org.kie.remote.services.jaxb.ServerJaxbSerializationProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.sun.xml.bind.v2.runtime.IllegalAnnotationException;
+import com.sun.xml.bind.v2.runtime.IllegalAnnotationsException;
+import com.sun.xml.bind.v2.runtime.Location;
+
+@ApplicationScoped
+public class DynamicJaxbContextManager {
+
+    private static final Logger logger = LoggerFactory.getLogger(DynamicJaxbContextManager.class);
+
+    private static AtomicInteger instanceCreated = new AtomicInteger(0);
+
+    public DynamicJaxbContextManager() {
+        if( ! instanceCreated.compareAndSet(0, 1) ) {
+            logger.debug("Instance {} of the {} created!", instanceCreated.incrementAndGet(), DynamicJaxbContext.class.getSimpleName() );
```

Comment:
in that case maybe increase the log level (info to be same as on 6.3.x branch) or even working otherwise might not be noticed.